Feature/events

msinovcic opened the pull request.

msinovcic committed:
Added models for Event functionality

Models Added:
Event
Start Page, End Page, Splash Page
EventSplashRel
QuestionEventRel
EventQuestionAnswer
ParticipantEventRel

hbarnardt committed:
Updated EventPages model

hbarnardt committed:
Added module type to module model

hbarnardt committed:
Added data migration for badges linked to events

hbarnardt committed:
Added end_processed field to event

hbarnardt committed:
Added task to award Champ badges after end deactivates

msinovcic committed:
seperated migrations

msinovcic committed:
Added partial event admin

msinovcic committed:
Merge branch 'feature/events' of https://github.com/praekelt/mobileu into feature/events

dbreedt committed:
Fixed problems in the EventAdmin page

msinovcic committed:
removed migrations

msinovcic committed:
Added new migrations

msinovcic committed:
Added event form validation

msinovcic committed:
Fixed naming error

msinovcic committed:
Changed event_participant_rel model

Added results receieved field to indicate if the user has seen their results.

msinovcic committed:
Added get_next_event_question method

msinovcic committed:
Fixed key error

msinovcic committed:
Added answer_event method

hbarnardt committed:
Updated datamigration add_event_badges

hbarnardt committed:
Change ,png to .png

hbarnardt committed:
Merge branch 'feature/events' of github.com:praekelt/mobileu into feature/events

msinovcic committed:
Added can_take_event method and fixed answer_event method

msinovcic committed:
Merge branch 'feature/events' of https://github.com/praekelt/mobileu into feature/events

msinovcic committed:
Fix

hbarnardt committed:
Updated can_take_event

msinovcic committed:
Changed  field name events -> event

msinovcic committed:
Merge branch 'feature/events' of https://github.com/praekelt/mobileu into feature/events

hbarnardt committed:
Added points earned from events to point recalculation

hbarnardt committed:
Fixed spelling mistake

dbreedt commented on content/tasks.py:
```diff
@@ -1,7 +1,43 @@
+        scores = {}
+        for participant in participants:
+            scores[participant.id] = EventQuestionRel.objects.filter(event=event, participant=participant, correct=True) \
+                .count()
```
20-25: Should be done with a single annotate
EventQuestionRel.objects.filter(...).annotate(score=Count("pk"))

dbreedt commented on content/tasks.py:
```diff
@@ -1,7 +1,43 @@
+        winners = Participant.objects.filter(id__in=winner_ids)
+
+        module = CourseModuleRel.objects.filter(course=event.course).first()
+        if "spot test" in event.name.lowercase():
```
There should be an event type that identifies the event not a name.

hbarnardt committed:
Updated event end processing task

msinovcic committed:
When adding questions to event, only published questions are available
Method can_take_event, added a condition to check if number of answered questions is less than total questions in the event
Fixed event dates validation issue

msinovcic committed:
When adding questions to event, only published questions are available
Method can_take_event, added a condition to check if number of answered questions is less than total questions in the event
Fixed event dates validation issue

msinovcic committed:
Merge branch 'feature/events' of https://github.com/praekelt/mobileu into feature/events

msinovcic committed:
Added event type to event model
